lms/djangoapps/courseware/tests/test_courses.py

```diff
@@ -80,6 +84,53 @@ def test_get_course_func_with_access(self, course_access_func, num_mongo_calls):
         with check_mongo_calls(num_mongo_calls):
             course_access_func(user, 'load', course.id)
 
+    def test_get_courses(self):
+        """
+        Verify that org filtering performs as expected, and that an empty result
+        is returned if the org passed by the caller does not match the designated
+        microsite org.
+        """
+        primary = 'primary'
+        alternate = 'alternate'
+
+        def _fake_get_value(value, default=None):
+            """Used to stub out microsite.get_value()."""
+            if value == 'course_org_filter':
+                return alternate
+
+            return default
+
+        user = UserFactory.create()
+
+        # Pass `emit_signals=True` so that these courses are cached with CourseOverviews.
+        primary_course = CourseFactory.create(org=primary, emit_signals=True)
+        alternate_course = CourseFactory.create(org=alternate, emit_signals=True)
+
+        self.assertNotEqual(primary_course.org, alternate_course.org)
+
+        unfiltered_courses = get_courses(user)
+        self.assertTrue(
+            any(course.org == alternate_course.org for course in unfiltered_courses)
+        )
+
```

Comment:
Should also add the following:
```
        self.assertTrue(
            any(course.org == primary_course.org for course in unfiltered_courses)
        )
```
Otherwise, we can't guarantee that primary_course is also returned.